### PR TITLE
meli: init module

### PR DESCRIPTION
**Pull request by colemickens**
### Description

This adds a `home-manager` module for `meli`: https://meli.delivery

See here for an example usage with Gmail: https://github.com/colemickens/nixcfg/commit/b6a5d4e6e93d4ea48a19efedf083d3d2cd6d7d4b

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/rycee/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by colemickens** [comment]
I don't think the build error is related to my change?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -84,6 +84,8 @@
 
 /modules/programs/mcfly.nix                           @marsam
 
+/modules/programs/meli.ni x                           @colemickens
+
```
Comment (berbiche) [inline .github/CODEOWNERS:88]: There's a space in the filename

File: modules/lib/maintainers.nix [hunk 2]
```diff
@@ -41,4 +41,10 @@
     github = "olmokramer";
     githubId = 3612514;
   };
+  colemickens = {
+    name = "Cole Mickens";
+    email = "cole.mickens@gmail.com";
+    github = "colemickens";
+    githubId = 327028;
+  };
```
Comment (berbiche) [inline modules/lib/maintainers.nix:49]: AFAIK this list is for maintainers that are not in Nixpkgs `maintainers/maintainer-list.nix` yet.

**Review by berbiche — APPROVED** [review]
LGTM.

Considering the  `accounts.email` module, would it be meaningful to provide a compability layer to allow users to set their accounts with that module instead of `meli.settings.accounts`?

**Comment by colemickens** [comment]
Updated for the maintainer/codeowners comments.

As for sharing the `accounts.email` abstraction, I'd definitely prefer to punt on that - I'm short on time and not super comfortable with email. To further defend the punt, it seems like something that could be added over time, and there would need to be consideration given to how this freeform `settings` works along  with something auto-derived from the `accounts.email`.

**Comment by berbiche** [comment]
Fair point.

I believe using `accounts.email` would allow for a concise and centralized way to configure similar programs.
Further, `home-manager` has a lot to gain from centralizing commonly shared options within a single module and allowing more advanced configuration with individual program settings through other modules.

Since you are also using the new formatters, the merging is handled automatically so a user could modify `programs.meli.settings.accounts` without worrying about overlaps and silent misconfiguration with `accounts.email`.

**Comment by teto** [comment]
how come I never  saw that MUA ? looks nice. Would be definitely cool to support the `accounts.email` feature otherwise there is little value in the module.

**Comment by colemickens** [comment]
Oh wow, mr bot, hi, please bring it back.

**Comment by sumnerevans** [comment]
@berbiche @teto were you wanting to get this integrated in with the existing `accounts.email` infrastructure?

**Comment by berbiche** [comment]
> @berbiche @teto were you wanting to get this integrated in with the existing `accounts.email` infrastructure?

Preferably IMO

**Pull request closed without merging**